File: RealtimeDatabase/Models/Responses/InfoResponse.cs
using System.Collections.Generic;

namespace RealtimeDatabase.Models.Responses
{
    public class InfoResponse : ResponseBase
    {
        public InfoResponse()
        {

        }

        public string[] PrimaryKeys { get; set; }

        public PropertyAuthInfo QueryAuth { get; set; }

        public AuthInfo CreateAuth { get; set; }

        public AuthInfo RemoveAuth { get; set; }

        public PropertyAuthInfo UpdateAuth { get; set; }
    }

    public class AuthInfo
    {
        public bool Authentication { get; set; }

        public bool Authorization => Roles != null;

        public string[] Roles { get; set; }

        public string FunctionName { get; set; }
    }

    public class PropertyAuthInfo : AuthInfo
    {
        public Dictionary<string, AuthInfo> Properties { get; set; }
    }
}

File: DemoClient/projects/ng-realtime-database/src/lib/models/response/info-response.ts
import {ResponseBase} from './response-base';

export interface InfoResponse extends ResponseBase {
  primaryKeys: string[];

  queryAuth: PropertyAuthInfo;
  createAuth: AuthInfo;
  removeAuth: AuthInfo;
  updateAuth: PropertyAuthInfo;
}

export interface AuthInfo {
  authentication: boolean;
  authorization: boolean;
  roles?: string[];
  functionName?: string;
}

export interface PropertyAuthInfo extends AuthInfo {
  properties: { [key: string]: AuthInfo };
}
